addons/flannel/0.21.5/install.sh

```diff
@@ -270,9 +270,59 @@ function remove_weave() {
 
     # all nodes
     ip link delete weave
+
     rm -rf /var/lib/weave
     rm -rf /etc/cni/net.d/*weave*
     rm -rf /opt/cni/bin/weave*
+
+    logStep "Verifying Weave removal"
+
+    local resources=("daemonset.apps/weave-net"
+                     "rolebinding.rbac.authorization.k8s.io/weave-net"
+                     "role.rbac.authorization.k8s.io/weave-net"
+                     "clusterrolebinding.rbac.authorization.k8s.io/weave-net"
+                     "clusterrole.rbac.authorization.k8s.io/weave-net"
+                     "serviceaccount/weave-net"
+                     "secret/weave-passwd")
+
+    for resource in "${resources[@]}"; do
+        if ! timeout 60 kubectl -n kube-system delete "$resource" --grace-period=60 >/dev/null 2>&1; then
+            logWarn "$resource is still present after the deletion attempt with a grace period of 1 minute."
+        fi
+    done
```

Comment:
do you really need a loop? wont that add much more time? also why mask the output? also you need `--ignore-not-found` otherwise it could fail if the resource is not already present which is not what you want

```suggestion
    if ! timeout 60 kubectl -n kube-system delete --ignore-not-found "${resources[@]}" --grace-period=60 ; then
        logWarn "Weave resources still present after the deletion attempt with a grace period of 1 minute."
    fi
```

Reply:
added some padding to the timeout as well